kks/cmd/test.py

```diff
@@ -94,15 +94,20 @@ def run_test(binary, input_file, output_file, env):
             click.secho(error_output)
         return False
 
-    with output_file.open('r') as output_f:
+    with output_file.open('rb') as output_f:
         expected_output = output_f.read()
 
-    actual_output = process.stdout.decode('utf-8')
+    actual_output = process.stdout
 
     if expected_output != actual_output:
         click.secho('WA', fg='red', bold=True)
-        print_diff(expected_output, actual_output, 'expected', 'actual')
-        click.secho()
+        try:
+            expected = expected_output.decode('utf-8')
+            actual = actual_output.decode('utf-8')
+            print_diff(expected, actual, 'expected', 'actual')
+            click.secho()
+        except UnicodeDecodeError:
+            click.secho('Binary outputs differ')
```

Comment:
Не понятно что это сообщение значит. Я ожидал что-то вида
```py
click.secho("Cant decode output as utf-8", fg='red')
```

Reply:
О, есть же фича про suggestions

```suggestion
            click.secho('Cant decode output as utf-8', fg='red')
```

Ты вроде можешь одной кнопкой поменять строку на мою

Reply:
Возможно, стоит как-то обозначить, что вывод не просто не декодируется, а ещё и отличается от ожидаемого

Reply:
Ой, да, тут же уже условие про `expected_output != actual_output`

Как насчет
```suggestion
            click.secho('Output differs, but cant be decoded as utf-8', fg='red')
```